src/lib.rs

```diff
@@ -339,3 +458,13 @@ pub fn metadata_run(manifest_path: Option<&Path>, deps: bool, features: Option<C
     let meta = serde_json::from_str(stdout)?;
     Ok(meta)
 }
+
+/// An iterator of Message.
+type MessageIterator<R> = serde_json::StreamDeserializer<'static, serde_json::de::IoRead<R>, Message>;
+
+// TODO: Should I use de::Read instead, to support deserializing from a slice?
```

Comment:
If it's strictly more powerful to use `de::Read`, then by all means

Reply:
On investigating this a bit, de::Read isn't very convenient. It's implemented on serde_json::de::IoRead, serde_json::de::SliceRead, etc... and not on the actual libstd. Besides, I have a hard time envisioning under which situation would someone try to read an array of cargo messages from a byte array instead of an std::io::Read...